app/queries/auto_verify_submission_mutator.rb

```diff
@@ -1,24 +1,29 @@
 class AutoVerifySubmissionMutator < ApplicationQuery
   include AuthorizeStudent
   include LevelUpEligibilityComputable
+  include DevelopersNotifications
 
   property :target_id, validates: { presence: { message: 'Blank Target Id' } }
 
   validate :can_be_auto_verified
   validate :ensure_submittability
 
   def create_submission
-    TimelineEvent.transaction do
-      submission = TimelineEvent.create!(target: target, passed_at: Time.zone.now)
-
-      students.map do |student|
-        student.timeline_event_owners.create!(timeline_event: submission, latest: true)
+    submission =
+      TimelineEvent.transaction do
+        TimelineEvent.create!(target: target, passed_at: Time.zone.now)
+          .tap do |submission|
+            students.map do |student|
+              student.timeline_event_owners.create!(timeline_event: submission, latest: true)
+            end
+
+            TimelineEvents::AfterMarkingAsCompleteJob.perform_later(submission)
```

Comment:
This `perform_later` should be outside of the transaction...

Reply:
Why? The delayed_job schedules jobs by creating entries in database, AFAIK this is how it is configured here? 
So if something goes wrong scheduled job will be rolled back with transaction.

Having it outside of transaction might cause that this will not be run if any error happen during job scheduling. 

BTW It has been inside transaction before my changes so I've kept this that way.

Reply:
I believe I'd explained this before over Discord. 😅 

It's basically to [make the process of switching to something like Sidekiq smoother](https://github.com/rails/rails/issues/26045).

> So if something goes wrong scheduled job will be rolled back with transaction.

I don't think we need to worry about the case where ActiveJob / delayed_job itself will fail. In any case, with non-DB backed ActiveJob adapters, this wouldn't be a thing anyway.

> BTW It has been inside transaction before my changes so I've kept this that way.

That's okay. Generally, when we modify any code, we try to leave it in a better state, even if it isn't directly related to the task at hand. 🙂

Reply:
> That's okay. Generally, when we modify any code, we try to leave it in a better state, even if it isn't directly related to the task at hand. 🙂

Yep :) 

> It's basically to make the process of switching to something like Sidekiq smoother.

I haven't connected that dots 😂 Doing now ;)